src/hats_import/catalog/arguments.py

```diff
@@ -137,6 +140,46 @@ def to_table_properties(
         } | self.extra_property_dict()
         return TableProperties(**info)
 
+    @classmethod
+    def reimport_from_hats(
+        cls, path: str | Path | UPath, output_dir: str | Path | UPath, **kwargs
+    ) -> ImportArguments:
+        """Generate the import arguments to reimport a HATS catalog with different parameters
+
+        Args:
+            path (str | Path | UPath): the path to the existing HATS catalog to reimport
+            output_dir (str | Path | UPath): the path to output the reimported catalog to
+            kwargs: any import arguments to update from the existing catalog
+
+        Returns:
+            A ImportArguments object with the arguments from the existing catalog, and any updates from kwargs
+        """
+
+        path = get_upath(path)
+
+        catalog = hats.read_hats(path)
+
+        column_names = catalog.schema.names if catalog.schema is not None else None
+        if column_names is not None:
+            column_names.remove(PARTITION_ORDER)
+            column_names.remove(PARTITION_DIR)
+            column_names.remove(PARTITION_PIXEL)
+
+        import_args = {
+            "catalog_type": catalog.catalog_info.catalog_type,
+            "ra_column": catalog.catalog_info.ra_column,
+            "dec_column": catalog.catalog_info.dec_column,
+            "input_path": path / DATASET_DIR,
+            "file_reader": ParquetReader(column_names=column_names),
+            "output_artifact_name": catalog.catalog_name,
+            "output_path": output_dir,
+            "use_healpix_29": True,
+            "add_healpix_29": False,
+        }
+
```

Comment:
Set `expected_total_rows` to original catalogs number of rows, just for a little added security.